### PR TITLE
event: use duration_cast more nicely in timer_impl

### LINKED ISSUE #4332: clean up management of durations in timer_impl
*Description*:
> In #4257 there was at one point a refactoring which moved TimerImpl::enableTimer(), which includes this:

```
      std::chrono::microseconds us = std::chrono::duration_cast<std::chrono::microseconds>(d);
      timeval tv;
      tv.tv_sec = us.count() / 1000000;
```

bringing it to the attention of @dnoe , who commented:

```
The conversion factors here could be more readable by using duration cast:

auto secs = std::chrono::duration_cast<std::chrono::seconds>(d);
auto usecs = std::chrono::duration_cast<std::chrono::microseconds>(d - secs);
tv.tv_secs = secs.count();
tv.tv_usecs = usecs.count();
```

I didn't want to rewrite the code as I was just moving it; and then I wound up unmoving it so it really isn't part of #4257, but I'm jotting this down as tech-debt.

### PR DESCRIPTION
**Pull request by roochowane**
fix #4332

### CODE REVIEW

File: source/common/event/timer_impl.cc [hunk 1]
```diff
@@ -22,11 +22,13 @@ void TimerImpl::enableTimer(const std::chrono::milliseconds& d) {
   if (d.count() == 0) {
     event_active(&raw_event_, EV_TIMEOUT, 0);
   } else {
-    // TODO(#4332): use duration_cast more nicely to clean up this code.
-    std::chrono::microseconds us = std::chrono::duration_cast<std::chrono::microseconds>(d);
+    auto sec = std::chrono::duration_cast<std::chrono::seconds>(d);
```
Comment (htuch) [inline source/common/event/timer_impl.cc]: Nit: `const`.

**Comment by htuch** [comment]
@ranchoooo how come you closed/deleted this?

**Comment by itayd** [comment]
(don't mind me, just testing if repokitteh issue is fixed)

**Pull request closed without merging**